### PR TITLE
Add private dns zone for platform.pagopa.it

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- Add private dns zone for `platform.pagopa.it` in order to allow the components in `vnet` and `vnet_integration` to resolve `api.platform.pagopa.it` with the apim ip.

### Motivation and context

The goal of this PR is to allow `pagopa-proxy` (but also other components) to resolve `api.platform.pagopa.it` with apim ip.

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/dns_private.tf [hunk 1]
```diff
@@ -41,3 +41,39 @@ resource "azurerm_private_dns_zone_virtual_network_link" "privatelink_redis_cach
 
   tags = var.tags
 }
+
+resource "azurerm_private_dns_zone" "apim_private_dns_zone" {
```
Comment (pasqualedevita) [inline src/dns_private.tf:45]: Better call platform_private_dns_zone

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
Top @infantesimone only change resource name and other references

File: src/dns_private.tf (hunk 1 above)
Comment (infantesimone) [reply src/dns_private.tf]: done

**Review by pasqualespica — APPROVED** [review]
🚀 great @infantesimone in this way all applications in to main `vnet` could call apim service directly 💪

**Pull request merged**